src/main/java/com/icthh/xm/gate/gateway/BeariringModeFilter.java

```diff
@@ -0,0 +1,60 @@
+package com.icthh.xm.gate.gateway;
+
+import static com.icthh.xm.commons.tenant.TenantContextUtils.getRequiredTenantKeyValue;
+
+import com.icthh.xm.commons.domain.idp.model.IdpPublicConfig.IdpConfigContainer.Features;
+
+import com.icthh.xm.commons.security.XmAuthenticationContext;
+
+import com.icthh.xm.commons.security.XmAuthenticationContextHolder;
+import com.icthh.xm.commons.tenant.TenantContextHolder;
+import com.icthh.xm.gate.security.oauth2.IdpConfigRepository;
+import com.netflix.zuul.ZuulFilter;
+import com.netflix.zuul.context.RequestContext;
+import javax.servlet.http.HttpServletRequest;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class BeariringModeFilter extends ZuulFilter {
```

Comment:
@olye0216 as I remember we have renamed the bearing feature. Please use the new name. 
But anyway seems this filter checks only stateful feature. So please choose correct name for the class.

Reply:
Renamed to IdpStatefulModeFilter